lms/djangoapps/teams/csv.py

```diff
@@ -306,6 +310,20 @@ def validate_user_assignment_to_team_and_teamset(self, row):
                 return False
         return True
 
+    def validate_compatible_enrollment_modes(self, user, team_name, teamset_id):
+        """
+        Validates that only students enrolled in a masters track are on a single team. Disallows mixing of masters
+        with other enrollment modes on a single team.
+        """
+        if(teamset_id, team_name) not in self.user_enrollment_by_team:
+            self.user_enrollment_by_team[teamset_id, team_name] = set()
+        self.user_enrollment_by_team[teamset_id, team_name].add(self.user_to_actual_enrollment_mode[user.id])
+        if 'masters' in self.user_enrollment_by_team[teamset_id, team_name] and \
```

Comment:
It was not obvious until I stepped through this that `user_enrollment_by_team` was actually tracking enrollment mode so maybe rename to `team_enrollment_modes`?

Reply:
I used to use the naming scheme you suggested, but for the purposes of this module, I was advised to use the x_by_y and stuck with it.